charmhelpers/core/hookenv.py

```diff
@@ -68,7 +68,10 @@ def unit_get(attribute):
     @wraps(func)
     def wrapper(*args, **kwargs):
         global cache
-        key = json.dumps((str(func), args, kwargs), sort_keys=True)
+        try:
+            key = json.dumps((str(func), args, kwargs), sort_keys=True)
+        except TypeError:
+            key = str((func, args, kwargs))
```

Comment:
We should probably modify the test at tests/core/test_hookenv.py +1176 to activate this piece of code?  i.e. pass something that isn't JSON serialisable and check that it still works as intended.

We also have the odd, but perhaps almost impossible to hit, potential issue of the json serialisation of an object being the same as a str-able set of parameters and thus different things ending up being cached.  Probably not something to worry about!

Reply:
I'll fix up the test on Monday. We can hold till then. At the time I thought this was a field emergency. But the workaround was acceptable.